### ISSUE TITLE
Possible memory leak in Input

### ISSUE THREAD

**Issue by davidbalbert** [ChimeHQ/SwiftTreeSitter#19]
I'm not familiar enough with the code to be sure, but I think the setter for Input.buffer leaks memory.

Existing version:

```swift
set {
    // if newValue != nil && internalBuffer != nil, we never deallocate internalBuffer
    if newValue == nil {
        internalBuffer?.deallocate()
    }

    internalBuffer = newValue
}
```

Possible fix:

```swift
set {
    internalBuffer?.deallocate()
    internalBuffer = newValue
}
```

**Comment by mattmassicotte** [comment]
Hmmmm, yes. I'm trying to figure out why I ever would have done it this way in the first place.

But, thanks so much for noticing!

I'm also wondering now if this can all be done with a value type...

**Comment by mattmassicotte** [comment]
Fixed by 1e8ad3c7d7f087c24467f2a8e0390405d18e2cb0
